static/js/stream_list.js

```diff
@@ -524,10 +524,7 @@ export function update_stream_sidebar_for_narrow(filter) {
         $stream_li.addClass("active-filter");
     }
 
-    if (stream_id !== topic_list.active_stream_id()) {
-        topic_zoom.clear_topics();
-    }
-
+    topic_zoom.clear_topics();
```

Comment:
I don't understand the reasoning for this commit. `build_stream_list` only calls `topic_list.clear()` if `!stream_groups.same_as_before`; whereas this commit message seems to assume that's unconditional.

Reply:
For the bug we are trying to fix, `stream_groups.same_as_before` is always `false` since the streams rearrange when new messages arrive in a steam for which we had no messages.

Also, I was testing if this is working correct by renaming the stream. I realize that it is a rare case, but we do call `build_stream_list` always in that case since `force_rerender` is `true`.

```js
export function rename_stream(sub) {
    // The sub object is expected to already have the updated name
    build_stream_sidebar_row(sub);
    update_streams_sidebar(true); // big hammer
}
```

Reply:
Clearly, I missed `stream_groups.same_as_before` check inside `build_stream_list` since it was always being completely called in the bugged cases. Let me look at this again.

Reply:
Updated.